core/src/test/java/org/aerogear/mobile/core/http/OkHttpServiceModuleTest.java

```diff
@@ -94,4 +96,36 @@ public void testCompleteHandlerCalledInErrorCase() {
         response.waitForCompletionAndClose();
         assertEquals(latch.getCount(), 0);
     }
+
+    @Test
+    public void testRedirectsShouldBeSuccessful() {
+        OkHttpClient client = new OkHttpClient.Builder()
+            .followRedirects(false)
+            .followSslRedirects(false)
+            .build();
+
+        HttpServiceModule module = new OkHttpServiceModule(client);
+
+        HttpRequest request = module.newRequest();
+        request.get("https://jigsaw.w3.org/HTTP/300/302.html");
```

Comment:
I see you are going to replace it with the mock server, just wanna leave a tip here. When I need to test some HTTP status I always use https://httpstat.us/